core/transaction-pool/graph/src/validated_pool.rs

```diff
@@ -182,18 +187,100 @@ impl<B: ChainApi> ValidatedPool<B> {
 		}
 	}
 
+	/// Resubmits transactions in the pool.
+	///
+	/// Removes and then submits passed transactions and all dependent transactions.
+	/// Transactions that are missing from the pool are not submitted.
+	pub fn resubmit(&self, mut transactions: HashMap<ExHash<B>, ValidatedTransactionFor<B>>) {
+		#[derive(Debug, Clone, Copy, PartialEq)]
+		enum Status { Future, Ready, Failed, Dropped };
+
+		let (initial_statuses, final_statuses) = {
+			let mut pool = self.pool.write();
+
+			// remove all passed transactions from the ready/future queues
+			// (this may remove additional transactions as well)
+			let mut initial_statuses = HashMap::new();
+			let mut txs_to_resubmit = Vec::with_capacity(transactions.len());
+			while !transactions.is_empty() {
```

Comment:
Alternatively we could do `transactions.into_iter()` and maintain a `HashSet<H256>` of already removed / resubmitted transactions? Have you considered that? I wonder if the code is simpler in such case.

Reply:
IMO this would require more `clone()` calls because we'll be unable to check if updated validation result exists for the transaction.

Reply:
(and as a result more calls to `HashMap::insert()` because we need to overwrite validation result with updated)